From 4714340c16624266c5225f3d74f02cc8230339bf Mon Sep 17 00:00:00 2001
From: Sahil Malhotra <sahil.malhotra@nxp.com>
Date: Thu, 7 Dec 2023 19:19:04 +0530
Subject: [PATCH] drivers: caam: update return type of caam_sm_free()

Update return type of caam_sm_free() from TEE_Result
to enum caam_status.

Upstream-Status: Backport [dc6563d7d6a4a1ab83e506aa101819a88efe418c]

Signed-off-by: Sahil Malhotra <sahil.malhotra@nxp.com>
Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>
Acked-by: Clement Faure <clement.faure@nxp.com>
---
 core/drivers/crypto/caam/include/caam_sm.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/drivers/crypto/caam/include/caam_sm.h b/core/drivers/crypto/caam/include/caam_sm.h
index 20acc487b5a..5dee4330085 100644
--- a/core/drivers/crypto/caam/include/caam_sm.h
+++ b/core/drivers/crypto/caam/include/caam_sm.h
@@ -61,7 +61,8 @@ caam_sm_set_access_perm(const struct caam_sm_page_desc *page_desc,
  *
  * @sm_page_descriptor    Secure Memory page
  */
-TEE_Result caam_sm_free(const struct caam_sm_page_desc *sm_page_descriptor);
+enum caam_status
+caam_sm_free(const struct caam_sm_page_desc *sm_page_descriptor);
 
 #else
 static inline enum caam_status caam_sm_init(struct caam_jrcfg *jrcfg __unused)
